neo/ProtocolSettings.cs

```diff
@@ -11,12 +12,31 @@ public class ProtocolSettings
         public string[] SeedList { get; }
         public uint SecondsPerBlock { get; }
 
-        public static ProtocolSettings Default { get; }
+        static ProtocolSettings _default;
 
-        static ProtocolSettings()
+        static bool UpdateDefault(IConfiguration configuration)
         {
-            IConfigurationSection section = new ConfigurationBuilder().AddJsonFile("protocol.json", true).Build().GetSection("ProtocolConfiguration");
-            Default = new ProtocolSettings(section);
+            var settings = new ProtocolSettings(configuration.GetSection("ProtocolConfiguration"));
+            return null == Interlocked.CompareExchange(ref _default, settings, null);
+        }
+
+        public static bool Initialize(IConfiguration configuration)
```

Comment:
Do we need this `Initialize()`? `UpdateDefault()` will be called automatically when accessing `Default`.

Reply:
Initialize is a public method intended to be called by host code. UpdateDefault is a private helper method that does the actual CompareExchange so that I didn't Duplicate code in the Initialize and Default getter code paths.

Reply:
This is very useful for testing. This way we can manage to set everything up before actually starting the other components. We could apply a similar solution to neo-plugins Settings: https://github.com/neo-project/neo-plugins/blob/a72c44813271efcafe0a4fcee2505fae3a3b4716/ApplicationLogs/Settings.cs#L9